[F-6] Register Page

PandaHajs opened the pull request:
Made the register page including:
- The register page
- Register logic
- Register form validation
- Redirects between the login page and register page

PandaHajs committed:
Initial Register Page

PandaHajs committed:
Removed unnecessary variable

PandaHajs committed:
Username

Added a username input during registration, also removed the forgotten console.log

PandaHajs committed:
Temporary logic and form validation

PandaHajs committed:
Remade register page styles

remade register page styles to look more like the login page, added redirects between login and register

PandaHajs committed:
Slight style changes

PandaHajs committed:
Removed comments

Aeri4a commented on apps/frontend/src/router/router.tsx:
```diff
@@ -71,4 +72,8 @@ export const router = createBrowserRouter([
         path: Route.NOT_FOUND,
         element: withGuard(<NotFound />),
     },
+    {
+        path: Route.REGISTER,
+        element: withGuard(<Register />),
```
Small suggestion - NOT_FOUND could be at the end.

Aeri4a commented on apps/frontend/src/auth/auth.utils.ts:
```diff
@@ -50,3 +55,19 @@ export const loginFormValidator = (form: LoginForm) => {
     return errors;
 };
+
+export const registerFormValidator = (form: RegisterForm) => {
+    const errors: RegisterValidationErrors = {};
+    const emailRegex = /^[\w-]{1,30}@([\w-]+\.)+[\w-]{2,4}$/g;
+    if (!emailRegex.test(form.email)) {
+        errors.email = 'Email is not in a valid format';
+    }
+    if (form.password.length < 8) {
+        errors.password = 'Password must be at least 8 characters';
+    }
+    if (form.username.length < 3) {
+        errors.username = 'Username must be at least 3 characters';
+    }
+
+    return errors;
+};
```
It is correct :). You asked how to do it in other way... the other solution is to handle errors as soon as the input changes and if we have any error set to true then we fail in validation method.

Aeri4a reviewed (dismissed):
Good job! :)
Optional comments if u would like to analyze

PandaHajs committed:
Errors now pop up in real time

register form errors now pop up in real time instead of only when the user clicks register
